src/components/AbstractModal/AbstractModal.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import Modal from 'react-modal';
+
+const AbstractModal = ({ modalId, className, required, modalIsOpen, children }) => {
+    return (
+        <Modal 
+            id={ modalId } 
+            className={ className }
+            shouldCloseOnOverlayClick={ !required }
+            shouldCloseOnEsc={ !required }
```

Comment:
same as above - I think this can be hardcoded, but as `true` for accessibility reasons. Some people need to be able to use the escape key to manipulate elements on the page, so if we set this to false they can no longer do that.

Reply:
done